Steve-O-Murphy analyzer apis documentation (#1594)
opensearch-project/documentation-website: _opensearch/rest-api/analyze-apis/terminology.md

---
layout: default
title: Analysis API Terminology
parent: Analyze APIs
grand_parent: REST API reference
nav_order: 1
---

# Terminology

The following sections provide descriptions of important text analysis terms. 

## Analyzers

> Analyzers instruct OpenSearch how to index and search text. Analyzers comprise three components: a tokenizer, zero or more token filters, and zero or more character filters. 

Reviewer: Suggestion only:
Grammatically, "Analyzers comprise three components..." I think is correct. But it sounds stiff and stops me. Does it sound awkward to skip to singular here?
"An analyzer comprises three components..." or even "An analyzer is composed of three components..."?
Even "Analyzers are composed of three components..." sounds better to me.

> Analyzers instruct OpenSearch how to index and search text. An analyzer is composed of three components: a tokenizer, zero or more token filters, and zero or more character filters. 

Reviewer 2: ```suggestion
Analyzers tell OpenSearch how to index and search text. An analyzer is composed of three components: a tokenizer, zero or more token filters, and zero or more character filters. 
```

> OpenSearch provides *built-in* analyzers. For example, the `standard` built-in analyzer converts text to lower case and breaks text into tokens based on word boundaries such as carriage returns and white space. The `standard` analyzer is also called the *default* analyzer and is used when no analyzer is specified in the text analysis request.

Reviewer 2: ```suggestion
OpenSearch provides *built-in* analyzers. For example, the `standard` built-in analyzer converts text to lowercase and breaks text into tokens based on word boundaries such as carriage returns and white space. The `standard` analyzer is also called the *default* analyzer and is used when no analyzer is specified in the text analysis request.
```

If needed, you can combine tokenizers, token filters, and character filters to create a *custom* analyzer.

#### Tokenizers

> Tokenizers break unstuctured text into tokens and maintain metadata about tokens such as their start and ending positions in the text.

Reviewer 2: ```suggestion
Tokenizers break unstuctured text into tokens and maintain metadata about tokens, such as their start and ending positions in the text.
```

#### Character filters

> Character filters examine textand perform translations such as changing, removing, and adding characters. 

Reviewer: Add space: textand

> Character filters examine text and perform translations such as changing, removing, and adding characters. 

Reviewer 2: ```suggestion
Character filters examine text and perform translations, such as changing, removing, and adding characters. 
```

#### Token filters

> Token filters modify tokens, performing operations such converting a token's characters to upper case, and adding or removing tokens. 

Reviewer: "...such as converting..."
Can probably ditch the comma before "...and adding..."

> Token filters modify tokens, performing operations such as converting a token's characters to upper case and adding or removing tokens. 

Reviewer 2: ```suggestion
Token filters modify tokens, performing operations such as converting a token's characters to uppercase and adding or removing tokens. 
```

## Normalizers

> Similar to analyzers, normalizers tokenize text, but return a single token only. Normalizers do not employ tokenizers, and make limited use of character and token filters, such as those that operate on one character at a time.

Reviewer: Can delete the comma before "...but return..."
Maybe: "Normalizers do not employ tokenizers; they make limited use of character and token filters, such as those that operate on one character at a time."

> By default, OpenSearch does not apply normalizers. To apply a normalizers, you must add them to your data before creating an index.

Reviewer 2: ```suggestion
By default, OpenSearch does not apply normalizers. To apply normalizers, you must add them to your data before creating an index.
```
